### ISSUE TITLE
Fields get a new parent in merged documents

### ISSUE THREAD

**Issue by fancycode** [pdfcpu/pdfcpu#778]
When documents containing fields are merged, a new parent for the fields of the source document is created and the source fields are added there:
https://github.com/pdfcpu/pdfcpu/blob/b89d7b1ab108d54a983c9653d47df07bbad5336e/pkg/pdfcpu/merge.go#L350-L354

This will cause problems if the (unmodified) field names are expected, e.g. for filling out later.

Example:
- Source document contains fields named `path1.field1` and `path2.field2`.
- Destination document contains field named `path1.field2`.

Expected result:
The fields in the merged document are named `path1.field1`, `path1.field2` and `path2.field2`.

Instead:
The fields in the merged document are named `path1.field2`, `1.path1.field1` and `1.path2.field2`.

**Comment by hhrutter** [comment]
Please describe usecase.
Merging forms is a complex operation.
I generally recommend exporting your fields once you are ready to fill your form.

**Comment by fancycode** [comment]
Here is one use case:
- The application has multiple small PDF documents containing fields.
- Depending on user input, some of them are merged together as output document.
- This output document is filled out automatically with values generated from the user input.

Now as the field names in the resulting document are different depending on which documents have been combined (and in which order), filling out the fields in the resulting document automatically is not possible.

**Comment by hhrutter** [comment]
tricky.. let me think about this.

**Comment by fancycode** [comment]
As written above, this unfortunately is not possible: The fillout data is generated by scripts for the different small PDF documents, i.e. not directly by the user.

However depending on the ordering and depending on which PDF documents are merged, the field names in the resulting document differ and don't match the generated data from the scripts.

**Comment by mdmcconnell** [comment]
I don't believe there is a complete generic way to solve this problem - what would you expect to happen if two fields in the files to be merged have the same fully qualified name?  Some ideas for partial solutions:

- Proceed as if all names will be unique, and fail with an error if they are not.  
- Place each form from the input files into its own field tree, perhaps named with the filename.  This is complicated when there is any reference to a field within the document, and can not deal with two copies of the same file are merged.
- If there are duplicate fully qualified field names, split the leaf into as many kids as necessary, with 1, 2, 3, ... names.  In this case two of the same file may be merged, but field names are no longer deterministic and there is the complication with references mentioned above.

**Comment by fancycode** [comment]
It's valid to have multiple fields with the same fully qualified name in a PDF. Radio buttons of course have to share the same name but it's also possible for regular text fields to have the same name. These will get the same contents when being filled out.

**Comment by mdmcconnell** [comment]
Interesting point - I assumed the intent of merging two forms containing fields with the same name would be to have two distinct fields in the merged document.  Maybe that's a bad assumption and the user would intend to have two instances of the same field.

**Comment by mdmcconnell** [comment]
Ultimately, these could be user-specified options in a complete form merge function.
